test/ng/compileSpec.js

```diff
@@ -4193,7 +4194,7 @@ describe('$compile', function() {
       dealoc(element);
       compile('<div><span my-component>');
       expect(componentScope).not.toBe(firstComponentScope);
-      expect(componentScope.$$isolateBindings).toBe(first$$isolateBindings);
+      expect(componentScope.$$isolateBindings).toEqual(first$$isolateBindings);
```

Comment:
I think that implementation detail is important. That whole point of moving `parseDirectiveBindings` out of the link phase was to avoid recreating the same objects over and over again (memory usage/gc). Currently this change is partially undoing that enhancement which is why this test has to be changed.

Can we just put an `if` statement around the `parseDirectiveBindings` call instead of changing this? Or would that break the fix for #10149?